Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,552 @@
+
+    //Temporary value until there UI to expose the fields.
+    static constexpr int s_numOfDefaultLocationsInLevel = 12;
+
+    void ViewBookmarkLoader::RegisterViewBookmarkLoaderInterface()
+    {
+        AZ::Interface<ViewBookmarkLoaderInterface>::Register(this);
+    }
+
+    void ViewBookmarkLoader::UnregisterViewBookmarkLoaderInterface()
+    {
+        AZ::Interface<ViewBookmarkLoaderInterface>::Unregister(this);
+    }
+
+    void ViewBookmarkLoader::SaveBookmarkSettingsFile()
+    {
+        auto registry = AZ::SettingsRegistry::Get();
+        if (registry == nullptr)
+        {
+            AZ_Warning("ViewBookmarkLoader", false, "Unable to access global settings registry. Editor Preferences cannot be saved");
+            return;
+        }
+
+        // Resolve path to user project folder
+        AZ::IO::FixedMaxPath editorBookmarkFilePath = AZ::IO::FixedMaxPath(AZ::Utils::GetProjectPath()) / "user/Registry/ViewBookmarks/";
+
+        editorBookmarkFilePath /= m_bookmarkfileName;
+
+        AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings;
+        dumperSettings.m_prettifyOutput = true;
+
+        AZStd::string bookmarkKey = "/" + m_bookmarkfileName;
+
+        dumperSettings.m_includeFilter = [&bookmarkKey](AZStd::string_view path)
+        {
+            AZStd::string_view o3dePrefixPath(bookmarkKey);
+            return o3dePrefixPath.starts_with(path.substr(0, o3dePrefixPath.size()));
+        };
+
+        AZStd::string stringBuffer;
+        AZ::IO::ByteContainerStream stringStream(&stringBuffer);
+        if (!AZ::SettingsRegistryMergeUtils::DumpSettingsRegistryToStream(*registry, "", stringStream, dumperSettings))
+        {
+            AZ_Warning(
+                "ViewBookmarkLoader", false, R"(Unable to save changes to the Editor Preferences registry file at "%s"\n)",
+                editorBookmarkFilePath.c_str());
+            return;
+        }
+
+        bool saved = false;
+        constexpr auto configurationMode =
+            AZ::IO::SystemFile::SF_OPEN_CREATE | AZ::IO::SystemFile::SF_OPEN_CREATE_PATH | AZ::IO::SystemFile::SF_OPEN_WRITE_ONLY;
+        if (AZ::IO::SystemFile outputFile; outputFile.Open(editorBookmarkFilePath.c_str(), configurationMode))
+        {
+            saved = outputFile.Write(stringBuffer.data(), stringBuffer.size()) == stringBuffer.size();
+        }
+
+        // Once written to the desired file remove the key from the settings registry
+        registry->Remove(bookmarkKey + "/");
+        AZ_Warning(
+            "ViewBookmarkLoader", saved, R"(Unable to save Local View Bookmark file to path "%s"\n)", editorBookmarkFilePath.c_str());
+    }
+
+    bool ViewBookmarkLoader::SaveBookmark(ViewBookmark bookmark, const StorageMode mode)
+    {
+        switch (mode)
+        {
+        case AzToolsFramework::ViewBookmarkLoaderInterface::StorageMode::Shared:
+            return SaveSharedBookmark(bookmark);
```

Comment:
nit: This should probably Assert or emit a warning - one question actually is it might be simplest to remove the concept of `StorageMode` from the `ViewBookmarkLoader` entirely and just have everything use the local stuff. I don't think the SharedBookmark code will ever be used here, so I think it's better to just pair back the API and have this maybe just be `LocalViewBookmarkLoader`. I know it's a bigger change but I think it'll simplify things a lot. (feel free to ping me to discuss if you want to talk it over).